### ISSUE TITLE
Begin/Rescue/End inside a block raises Style/RedundantBegin

### ISSUE THREAD

**Issue by Rataah** [rubocop-hq/rubocop#6239]
Incorrect behavior of cop Style/RedundantBegin.
Maybe linked to #6192  but the other one only concerns stabby lambda

--------

## Expected behavior

```ruby
Object.new.tap do |obj|
  begin
    puts obj.object_id
  rescue StandardError => error
    p error
  end
end
```

should not raise Style/RedundantBegin

## Actual behavior

```
lib/rubocop_redundant_begin.rb:2:3: C: Style/RedundantBegin: Redundant begin block detected.
  begin
  ^^^^^
```

## RuboCop version

```
rubocop -V
0.58.2 (using Parser 2.5.1.2, running on ruby 2.4.2 x64-mingw32)
```

**Comment by Drenmi** [comment]
Please elaborate what's incorrect about this. In the case of #6192 it was clear, because stabby lambdas don't support the new syntax, but `#tap` does.
